modules/core/src/main/java/org/apache/ignite/internal/processors/platform/PlatformProcessorImpl.java

```diff
@@ -700,6 +704,17 @@ private void loggerLog(int level, String message, String category, String errorI
 
                 return;
             }
+
+            case OP_NODE_VERSION: {
+                IgniteProductVersion productVersion = ignite().cluster().node().version();
+                writer.writeByte(productVersion.major());
```

Comment:
Duplicates the code above. Please extract to `PlatformUtils.writeProductVersion` method.